root/tree/entrylist/regression_intoverflow.cxx

```diff
@@ -0,0 +1,80 @@
+#include "RtypesCore.h"
+#include "TChain.h"
+#include "TEntryList.h"
+#include "TFile.h"
+#include "TSystem.h"
+#include "TTree.h"
+
+#include "gtest/gtest.h"
+
+void FillTree(const char *filename, const char *treeName, ULong64_t nentries)
+{
+    TFile f{filename, "RECREATE"};
+    TTree t{treeName, treeName};
+
+    ULong64_t x;
+    t.Branch("x", &x);
+
+    for (ULong64_t i = 0; i < nentries; ++i)
+    {
+        x = i;
+        t.Fill();
+    }
+
+    t.Write();
+    f.Close();
+}
+
+/*
+Regression test for https://github.com/root-project/root/issues/11026
+
+Creates one tree with exactly (std::numeric_limits<int>::max()/100) entries.
+Then builds a TChain with 101 such trees, so that the total number of entries is
+higher than the maximum value of int. The chain is connected in lockstep with a
+TEntryList which has one sub TEntryList per tree in the chain.
+
+The test finally calls `TEntryList::GetEntryAndTree` for the entry index right
+after the maximum int value. This triggered the int overflow bug described in
+the linked issue.
+*/
+TEST(TEntryList, EntryGreaterThanMaxInt)
+{
+    constexpr int limit = std::numeric_limits<int>::max();
```

Comment:
Should not rely on transitive inclusion and we should perhaps add `#include <limits>`.

Reply:
Good catch, I had it in the original reproducer and somehow missed to copy it :(